sklearn/manifold/tests/test_t_sne.py

```diff
@@ -263,3 +389,90 @@ def test_chebyshev_metric():
     X = random_state.randn(5, 2)
     tsne.fit_transform(X)
 
+
+def test_no_sparse_on_barnes_hut():
+    """No sparse matrices allowed on Barnes-Hut"""
+    random_state = check_random_state(0)
+    X = random_state.randn(100, 2)
+    X[(np.random.randint(0, 100, 50), np.random.randint(0, 2, 50))] = 0.0
+    X_csr = sp.csr_matrix(X)
+    tsne = TSNE(n_iter=199, method='barnes_hut')
+    assert_raises_regexp(TypeError, ".*sparse.*standard.*", tsne.fit_transform,
+                         X_csr)
+
+
+def test_no_4D_on_barnes_hut():
+    """No sparse matrices allowed on Barnes-Hut"""
```

Comment:
The docstring is does not reflect the test. As the test method name is explicit there is no need for a docstring anyway.